core/src/inputs/stack.rs

```diff
@@ -70,3 +68,23 @@ impl IntoIterator for StackInputs {
         self.values.into_iter()
     }
 }
+
+impl Serializable for StackInputs {
+    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
+        self.values.write_into(target)
+    }
+}
+
+impl Deserializable for StackInputs {
+    fn read_from(bytes: &mut ByteReader) -> Result<Self, SerializationError> {
+        let values = Vec::read_from(bytes)?;
+        Ok(Self { values })
+    }
+}
+
+impl winter_utils::Serializable for StackInputs {
+    fn write_into<W: winter_utils::ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.values.len() as u64);
+        Felt::write_batch_into(&self.values, target);
+    }
+}
```

Comment:
We should make sure these serialize in the same way as it may have unintended consequences elsewhere.

Can we get rid of non-winterfell serialization/deserialization here? It is not clear why we need it yet.

Reply:
It is indeed an open question on what to do with the divergent serialization. `winter_utils::Serializable` is a requirement of `Air` for some of its attributes (such as `PublicInputs`), so this will be indeed the minimum requirement for the whole system to work. I'll split this extra serde from this PR